Client/mods/deathmatch/logic/CClientSoundManager.cpp

```diff
@@ -123,14 +123,22 @@ void CClientSoundManager::SetDimension(unsigned short usDimension)
     m_usDimension = usDimension;
 }
 
-CClientSound* CClientSoundManager::PlaySound2D(const SString& strSound, bool bIsURL, bool bLoop, bool bThrottle)
+CClientSound* CClientSoundManager::PlaySound2D(const SString& strSound, bool bIsURL, bool bIsRawData, bool bLoop, bool bThrottle)
 {
     CClientSound* pSound = new CClientSound(m_pClientManager, INVALID_ELEMENT_ID);
     if (bIsURL)
     {
         pSound->PlayStream(strSound, bLoop, bThrottle);
         return pSound;
     }
+    else if (bIsRawData)
+    {
+        size_t size = strSound.size();
+        void* pMemory = new char[size];
+        memcpy(pMemory, strSound.data(), size);
+        if (pSound->Play((void*)pMemory, size, bLoop))
```

Comment:
Just dropping this note here so that I (or another reviewer) doesn't forget about it.

Blindly trusting the input in this way scares me. Without looking at all into what `pSound->Play`, I hope it first uses some kind of decoder to verify that the sound is actually valid.

Reply:
I assumed that BASS_StreamCreateFile and BASS_MusicLoad do handle input properly but then again I can't be sure since the BASS library isn't open source.

Reply:
Both of those BASS functions can return a BASS_ERROR_FILEFORM error code if the file format is not recognised, so I agree with BASS verifying the memory buffer contents being a reasonable assumption. It'd definitely be scary if it didn't.